components/server/src/workspace/context-parser.ts

```diff
@@ -96,12 +96,24 @@ export interface URLParts {
  * and expects "othercontext" to be parsed and passed back.
  */
 export interface IPrefixContextParser {
+    // readonly property to see if prefix context will specify an IDE
+    specifiedIDE: boolean;
```

Comment:
Update interface `IPrefixContextParser` so that other prefix context parsers that may be added in the future will not break this PR

Reply:
I agree that leaking info about a single context on generic interface is not really great. Why don't we only check for `WithReferrerContext`?

Reply:
@akosyakov  `WithReferrerContext` works, but if someone add a new prefix like `/#editor` () in the future and don't know this experiment, the prefix will not work for first-time users. That's why we extend `IPrefixContextParser`.

If you don't think so, check with `WithReferrerContext` is also good for me

Reply:
When we can discuss it now? And we get away then with a super context like `IDEAwareContext = WithReferrerContext | WithEditorContext´ and one is check which delegates both? 

I don't think a result should be propagated via boolean properties.